src/components/forms/FormWorkflow.tsx

```diff
@@ -0,0 +1,124 @@
+
+import ConfigError from "../settings/ConfigError";
+import ConfigModal from "../settings/SettingsLMSTab/ConfigModal";
+import FormContextProvider, {
+  useFormContext,
+    FormFields,
+    FormFieldValidation,
+  } from "./FormContext";
+import FormReducer, { initializeForm } from "./data/reducer";
+
+export type FormWorkflowButtonConfig = {
+  buttonText: string;
+  onClick: (event: SyntheticEvent, formFields: FormFields) => Promise<void>;
+};
+
+type DynamicComponent = React.FunctionComponent | React.ComponentClass;
+
+export type FormWorkflowStep = {
+  index: number;
+  stepName: string;
+  formComponent: DynamicComponent;
+  validations: FormFieldValidation[];
+  nextButtonConfig: FormWorkflowButtonConfig;
+};
+
+// TODO: Standardize this
+export interface FormWorkflowConfigProps {}
+
+export type FormWorkflowConfig = {
+  steps: FormWorkflowStep[];
+  configProps: FormWorkflowConfigProps;
+  getCurrentStep: () => FormWorkflowStep;
+};
+
+export type FormWorkflowProps<FormData> = {
+  formWorkflowConfig: FormWorkflowConfig;
+  onClickOut: (code: string) => null;
+  onSubmit: (formData: FormData) => null;
+  formData: FormData;
+};
+
+// Modal container for multi-step forms
+function FormWorkflow<FormData>({
+  formWorkflowConfig,
+  onClickOut,
+  onSubmit,
+  formData,
+}: FormWorkflowProps<FormData>) {
+  const { hasErrors } = useFormContext();
+  const [errorIsOpen, openError, closeError] = useToggle(false);
+  const [savedChangesModalIsOpen, openModal, closeModal] = useToggle(false);
+  const step: FormWorkflowStep = formWorkflowConfig?.getCurrentStep();
+  const FormComponent: DynamicComponent = step?.formComponent;
+
+  const [formFieldsState, dispatch] = useReducer(
+    FormReducer,
+    initializeForm(
+      {},
+      { formFields: formData as FormFields, validations: step.validations }
+    )
+  );
+
+  return (
+    <>
+      <FormContextProvider
+        dispatch={dispatch}
+        formContext={formFieldsState || {}}
+      >
+        <>
+          <ConfigError isOpen={errorIsOpen} close={closeError} />
+          {/* TODO: Rename this Modal Component when converting to .tsx */}
+          <ConfigModal
+            isOpen={savedChangesModalIsOpen}
+            close={closeModal}
+            onClick={onClickOut}
+            saveDraft={onSubmit}
+          />
+
+          {formWorkflowConfig.steps && (
+            <Stepper activeKey={step}>
+              {formWorkflowConfig.steps.map((stepConfig) => (
+                <Stepper.Step
+                  eventKey={stepConfig.index}
+                  title={stepConfig.stepName}
+                />
+              ))}
+            </Stepper>
+          )}
+          <span>
+            <Form>
+              {step && FormComponent && (
+                <>
+                  <FormComponent {...formWorkflowConfig.configProps} />
+                  <span className="d-flex">
+                    {/* TODO: Fix typescript issue with Paragon Button */}
```

Comment:
We'll either need to find type definitions from the Paragon library Button component, or include our own.
<img width="800" alt="image" src="https://user-images.githubusercontent.com/322346/213583041-a5935bbc-d778-4e77-b914-87098337c2b1.png">

Reply:
Feel free to make an upstream Paragon contribution to add types to the `Button` component! The only existing component we've cut over to TypeScript thus far is `Bubble`, which defines an `interface` for the `BubbleProps`. While `BubbleProps` is exported in `./src/Bubble/index.tsx`, it's not exported by `./src/index.js`; without the `interface` exported there, would an MFE be able to pick up the types for the component? For example, if you define an `interface` for `Button`, does that interface need to be exported by Paragon's `./src/index.js`?